stdlib/public/runtime/Metadata.cpp

```diff
@@ -3945,6 +3949,33 @@ void Metadata::dump() const {
   printf("Type Context Description: %p.\n", getTypeContextDescriptor());
   printf("Generic Args: %p.\n", getGenericArgs());
 }
+
+template <>
+LLVM_ATTRIBUTE_USED
+void TypeContextDescriptor::dump() const {
+  printf("TargetTypeContextDescriptor.\n");
```

Comment:
Looks awesome! My only question is why you are using this in only some of the places.

Reply:
I banged these out while debugging a problem and figured I should add them in. We should definitely make more though.

Reply:
No that isn't what I meant. I meant in some parts you are using "this->" and others you aren't.

Reply:
Oh, `this`! Hah. The compiler complained about not finding those fields without it. I don't really understand why.

Reply:
From talking with JoeG it is due to dependent member lookup and c++ being what it is. This looks great to me!

Reply:
Thanks!